athena-elasticsearch/src/main/java/com/amazonaws/connectors/athena/elasticsearch/ElasticsearchMetadataHandler.java

```diff
@@ -179,9 +194,6 @@ public ListTablesResponse doListTables(BlockAllocator allocator, ListTablesReque
             catch (IOException error) {
                 throw new RuntimeException("Error retrieving indices: " + error.getMessage(), error);
             }
```

Comment:
when reusing the client I think you need to consider 'destroying' the client if you encounter an exception from the client otherwise you might not know if the client is 'bad'. The hbase connector has some handling for this, we basically wrapp the client to catch and auto-retry with a new client: https://github.com/awslabs/aws-athena-query-federation/blob/master/athena-hbase/src/main/java/com/amazonaws/athena/connectors/hbase/connection/HBaseConnection.java

Reply:
Client health is checked in CacheableAwsRestHighLevelClient before returning it from the cache. If the client fails the test, it is discarded from the cache.